src/state_manager/mod.rs

```diff
@@ -1297,6 +1298,22 @@ where
                             ?actual_receipt,
                             "state mismatch"
                         );
+                        // The flag is `false` by default, to enable it, set env var FOREST_VERBOSE_STATE_DIFF=1 or FOREST_VERBOSE_STATE_DIFF=true
+                        let should_print_state_diff =
+                            match std::env::var("FOREST_VERBOSE_STATE_DIFF") {
+                                Ok(value) if value == "1" || value.to_lowercase() == "true" => true,
+                                _ => false,
+                            };
```

Comment:
I greatly prefer the PR without this code. It's unclear to me why you added this.

Reply:
Removed